Mathlib/Topology/KrullDimension.lean

```diff
@@ -0,0 +1,23 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Fangming Li
+-/
+import Mathlib.Order.KrullDimension
+import Mathlib.Topology.Irreducible
+
+/-!
+# The Krull dimension of a topological space
+
+The Krull dimension of a topological space is the order theoretic Krull dimension applied to the
+collection of all its subsets that are closed and irreducible. Unfolding this definition, it is
+the length of longest series of closed irreducible subsets ordered by inclusion.
+-/
+
+/--
+The Krull dimension of a topological space is the supremum of lengths of chains of
+closed irreducible sets.
+-/
+noncomputable def topologicalKrullDim (T : Type _) [TopologicalSpace T] :
+    WithBot (WithTop ℕ) :=
+  krullDim { s : Set T | IsIrreducibleClosed s }
```

Comment:
Should we define `IrreducibleCloseds` as a subtype and set up an order on `IrreducibleCloseds` and define `topologicalKrullDim` as `krullDim( irreducibleCloseds T)`?

Reply:
Whether or not we do this, we have `IsIrreducible` and `IsClosed` so I don't see any harm in `IsIrreducibleClosed` for now. @erdOne do you have an opinion on this? If we're happy with the definition as it is, this is probably ready to go now. But if we want to change, probably now is a good time to do it.

Reply:
I added that definition.